scipy/spatial/_spherical_voronoi.py

```diff
@@ -16,6 +16,7 @@
 import scipy
 from . import _voronoi
 from scipy.spatial import cKDTree
+from ._voronoi_pythran import sort_vertices_of_regions
```

Comment:
Don't we need a guard for the import of the Cython extension? Locally with this feature branch I'm getting:

```python
scipy/spatial/__init__.py:99: in <module>
    from ._spherical_voronoi import SphericalVoronoi
scipy/spatial/_spherical_voronoi.py:17: in <module>
    from . import _voronoi
E   ImportError: cannot import name '_voronoi' from partially initialized module 'scipy.spatial' (most likely due to a circular import) (/Users/treddy/github_projects/scipy/build/testenv/lib/python3.8/site-packages/scipy/spatial/__init__.py)

```

Even if I fix that with:

```diff
--- a/scipy/spatial/_spherical_voronoi.py
+++ b/scipy/spatial/_spherical_voronoi.py
@@ -14,9 +14,11 @@ Spherical Voronoi Code
 import warnings
 import numpy as np
 import scipy
-from . import _voronoi
 from scipy.spatial import cKDTree
-from ._voronoi_pythran import sort_vertices_of_regions
+try:
+    from . import _voronoi
+except ImportError:
+    from ._voronoi_pythran import sort_vertices_of_regions
 

```

I still see a few test failures for `test_spherical_voronoi.py` locally.

Reply:
The current version is incorrect but the solution should be a bit different: we want to produce a `_voronoi.so` extension, not `_voronoi_pythran.so`. That way the import can simply be `from . import _voronoi`, and it will pick up a compiled version (the Python code in this file simply doesn't know if it was Cython or Pythran that was used at build time).

Reply:
Right, I was just guarding against the immediate error--`import ._voronoi_pythran as _voronoi` in the `except` block might make it explicit that two possible implementations exist.